p2p/rssh.sh

```diff
@@ -0,0 +1,412 @@
+
+devicename is the remote.it name you gave to this device's SSH connection.
+
+If your device name has spaces in it, surround "username@device name" with quotes.
+
+Verbose output
+
+To get more information about the internal operation of the script, use one or two -v switches, e.g.
+
+./rssh.sh -v username@device-name
+
+./rssh.sh -v -v username@device-name
+
+Clearing the cache
+
+To clear out all cached data (port assignments, device list)
+
+./rssh.sh -r
+
+Connection times may be a little slower and you may get connection security warnings after doing this 
+until all services have been connected to once.
+
+To cleanup (reset login authorization and active port assignments)
+
+./rssh.sh -c
+
+After running this, you will need to log in again.
+
+How the script works
+
+The script starts by logging into the remote.it server to obtain a login token.  All API calls are 
+documented here:
+
+https://docs.remote.it/api-reference/overview
+
+The user token is sent to the Service List API call in order to retrieve the full device list 
+associated with this account.
+
+From there we parse the JSON output of the device list and find the entry corresponding to the device 
+name you gave.  We find the UID (JSON ["deviceaddress"]) for this entry and use this in conjunction 
+with the remote.it daemon (connectd) in client mode to initiate a peer to peer connection.
+
+/usr/bin/connectd -c <base64 of username> <base64 of password> <UID> T<portnum> <Encryption mode> <localhost address> <maxoutstanding>
+
+-c = client mode
+<base64 of username> = remote.it user name, base64 encoded
+<base64 of password> = remote.it password, base64 encoded
+<UID> = remote.it UID (deviceaddress metadata) for this remote.it Service
+<portnum> = port to use on localhost address
+<Encryption mode> = 1 or 2
+<localhost address> = 127.0.0.1
+<maxoutstanding> = 12
+
+Example:
+/usr/bin/connectd -c ZmF1bHReaX5lMTk9OUB5YWhvby5jb20= d5VhdmVkFjAxWg== 80:00:00:0F:96:00:01:D3 T33000 1 127.0.0.1 12
+
+Now you have a listener at 127.0.0.1:33000 that allows a connection to your remote device's VNC service.
+
+The command line ssh client is launched and you are greeted with a request for your SSH password.  
+Until the port assignment values are cached, you may see SSH security warnings.
+
+EOF
+#
+printf "\n%s\n\n\n" "$man_text"
+}
+
+#
+# Print Usage
+#
+usage()
+{
+        echo "Usage: $0 [-v (verbose)] [-v (maximum verbosity)] [-l(ist services only)] [-c(leanup)] [-r(eset to default)] [-m(an page)] [-i (use PEM key for SSH login)] [-h (this message)] [user@]<devicename> [passed on to ssh]" >&2
+        echo "     [optional] must specify device name." >&2
+        echo "Version $VERSION Build $MODIFIED" >&2
+        exit 1 
+}
+
+
+###############################
+# Main program starts here    #
+###############################
+#
+# Create the config directory if not there
+#
+echo "remote.it rssh.sh Version $VERSION $MODIFIED"
```

Comment:
Should we echo this on every call to the command line? Seems excessive and atypical

Reply:
I can have that only when using -v if you like.